Agents/HistoricalNTUEnergyAgent/README.md

```diff
@@ -0,0 +1,190 @@
+- One data file for energy readings in Excel format needs to be named `NTU_Energy_Consumption.xlsx` and included in the `config` folder. An example data file can be found at HistoricalNTUEnergyAgent\data\NTU_Energy_Consumption_Example.xlsx.
+
+### Mapping files
+The mapping files define how data received from the API is connected
+to the knowledge graph (KG). Specifically, each JSON key in the readings 
+represents a specific measure that needs to be represented by an IRI, if it should be saved in the database.
+The mapping is achieved in this package by using one property file per group. Each property file contains one line per
+JSON key that should be linked to an IRI, e.g. like:
+```
+N1_3_P_KW=http:/example/N1_3_P_KW
+```
+If the IRI is left empty (`N1_3_P_KW=` in the example), i.e. because there is no instance that represents the measure yet,
+it will be automatically created when the agent is run for the first time. This automatically generated URI will have the
+following form:
+```
+[prefix]/[key]_[UUID]
+```
+where the `[prefix]` is hardcoded into the `HistoricalNTUEnergyAgent` class in a public, static field called `generatedIRIPrefix`
+which is based on the time-series client namespace, `[key]` is the JSON key the URI is generated for, and `[UUID]` is a
+randomly generated UUID.
+
+### Agent properties
+The agent property file only needs to contain a single line:
+```
+ntuenergy.mappingfolder=HISTORICALNTUENERGY_AGENT_MAPPINGS
+```
+where `HISTORICALNTUENERGY_AGENT_MAPPINGS` is the environment variable pointing to the location of a folder containing JSON key to IRI mappings.
+An example property file can be found in the `config` folder under `agent.properties`.
+
+### Excel connector properties
+The Excel connector properties contain the number of columns/keys for each of the Excel files.
+It should contain the following keys:
+- `numOfEnergyKeys` the number of columns/keys in the energy readings Excel file.
+More information can be found in the example property file `xlsxconnector.properties` in the `config` folder.
+
+The next section will explain the detailed instructions to run the agent.
+
+## 2.2 Docker Setup
+
+### 2.2.1 Build Docker Image
+The NTUEnergyAgent is set up to use the Maven repository. You'll need to provide your credentials in single-word text files located like this:
+```
+./credentials/
+    repo_username.txt
+    repo_password.txt
+```
+repo_username.txt should contain your Github username, and repo_password.txt your Github [personal access token](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token),
+which must have a 'scope' that [allows you to publish and install packages](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-apache-maven-registry#authenticating-to-github-packages).
+
+Then build image with:
+```
+docker build -t historical-ntuenergy .
```

Comment:
i had to build historical-ntuenergy-agent:1.0.0 to consistent with the image name in the stack manager config

Reply:
Fixed.